src/features/thai/thaiLigatures.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * Apply Thai Ligatures feature to tokens
+ */
+
+import { ContextParams } from '../../tokenizer.js';
+import applySubstitution from '../applySubstitution.js';
+
+/**
+  * Update context params
+  * @param {any} tokens a list of tokens
+  * @param {number} index current item index
+  */
+function getContextParams(tokens, index) {
+    const context = tokens.map(token => token.activeState.value);
+    return new ContextParams(context, index || 0);
+}
+
+/**
+  * Apply Thai required glyphs composition substitutions
+  * @param {ContextRange} range a range of tokens
+  */
+function thaiLigatures(range) {
+    const script = 'thai';
+    let tokens = this.tokenizer.getRangeTokens(range);
+    let contextParams = getContextParams(tokens, 0);
+    contextParams.context.forEach((glyphIndex, index) => {
```

Comment:
I don't generally recommend foreach as function calls can be very expensive when done in a loop, often a better solution is for-in loops or for loops. But we can fix this later.